gui/static/js/components/scenarios.js

```diff
@@ -85,12 +85,35 @@ export default class ScenariosItem extends React.Component {
         var _this = this;
         var file = event.target.files[0];
         var reader = new FileReader();
-        reader.onload = function(event) {
-            var scenario = JSON.parse(event.target.result).items;
-            _this.setState({items: scenario});
-        };
+        var li = [];
+        var objectString;
+        _this.state.numBots = 0;
+        reader.onload = function(fileLoadedEvent) {
+            objectString = fileLoadedEvent.target.result;
+            var jsonArray = JSON.parse(objectString);
+            var regexPosition = new RegExp("\[\d,\d]");
+            jsonArray.forEach(function(object) {
+                //checks that bot contains position and angle
+                if (regexPosition.test(object.position) && object.type == "simulator.simbot" && object.position != "undefined" && object.angle != "undefined")  {
+                    _this.state.numBots++;
```

Comment:
might not want to modify the state if you don't know if the file is valid

Reply:
fixed! I only change the state if the file passes all the checks